ha/core/event_manager/event_manager.py

```diff
@@ -320,16 +322,10 @@ def subscribe(self, component: SUBSCRIPTION_LIST, events: List[SubscribeEvent])
                 subscription_key = EVENT_MANAGER_KEYS.SUBSCRIPTION_KEY.value.replace("<component_id>", component)
                 if event.functional_types:
```

Comment:
How do keys get added if event.functional_types is not set here ?

Reply:
Default functional_type 'all' is used, in case no functional_type specified in SubscribeEvent.
So here it will be always set.